### PR TITLE
Cleared switching bug, and checked all possibilities

### LINKED ISSUE #305: Switching between examples causes problems due to data listeners.
@domoritz 

The previous state is preserved which causes the api to throw errors.

### PR DESCRIPTION
**Pull request by ganeshpatro321**
Fixes #305 .

Clearing the state before switching views.
@domoritz Please review.

### CODE REVIEW

File: src/components/data-viewer/renderer.tsx [hunk 1]
```diff
@@ -65,7 +65,19 @@ export default class DataViewer extends React.Component<Props, State> {
...
+    if (this.state.selectedData === '') {
+      const datasets = this.getDatasets();
+      if (datasets.length) {
+        this.setState({
+          selectedData: datasets[1],
+        });
+      }
+    }
+    if (this.state.selectedData !== '' && this.state.selectedData !== prevState.selectedData) {
       if (prevState.selectedData) {
```
Comment (domoritz) [inline src/components/data-viewer/renderer.tsx]: This line is not necessary anymore, right?

File: src/components/data-viewer/renderer.tsx [hunk 2]
```diff
@@ -65,7 +65,19 @@ export default class DataViewer extends React.Component<Props, State> {
...
-    if (this.props.view !== prevProps.view || this.state.selectedData !== prevState.selectedData) {
+    if (this.props.view !== prevProps.view) {
+      this.setState(initialState);
+      return;
+    }
+    if (this.state.selectedData === '') {
+      const datasets = this.getDatasets();
+      if (datasets.length) {
+        this.setState({
+          selectedData: datasets[1],
```
Comment (domoritz) [inline src/components/data-viewer/renderer.tsx]: Can we delete the code is component did mount now?

Comment (ganeshpatro321) [reply src/components/data-viewer/renderer.tsx]: No, we need it.

File: src/components/data-viewer/renderer.tsx (hunk 1 above)
Comment (ganeshpatro321) [reply src/components/data-viewer/renderer.tsx]: Which line are you talking about?

Comment (domoritz) [reply src/components/data-viewer/renderer.tsx]: 81

File: src/components/data-viewer/renderer.tsx (hunk 2 above)
Comment (domoritz) [reply src/components/data-viewer/renderer.tsx]: Can you explain?

File: src/components/data-viewer/renderer.tsx (hunk 1 above)
Comment (ganeshpatro321) [reply src/components/data-viewer/renderer.tsx]: Yeah, line 81 can be removed

Comment (ganeshpatro321) [reply src/components/data-viewer/renderer.tsx]: This should be present.

**Comment by domoritz** [comment]
Manually merged

**Pull request closed without merging**